src/fluree/db/dbfunctions/internal.cljc

```diff
@@ -579,3 +579,34 @@
     (let [base (.nextDouble (java.util.Random. instant))
           num  (int (Math/floor (* base max')))] num)
     (catch* e (function-error e "rand" instant max'))))
+
+(defn cas
+  "Returns new-val if existing-val is equal to compare-val, else throws exception"
+  [?ctx compare-val new-val]
+  (go-try
+    (let [{:keys [sid pid db]} ?ctx
+          p-name      (dbproto/-p-prop db :name pid)
+          _           (when-not sid
+                        (throw (ex-info (str "Unable to execute cas - subject id could be determined. Cas values: " compare-val new-val)
+                                        {:status 400
+                                         :error  :db/validation-error})))
+          _           (when-not p-name
+                        (throw (ex-info (str "Unable to execute cas - predicate could be determined. Cas values: " compare-val new-val)
+                                        {:status 400
+                                         :error  :db/validation-error})))
+          _           (when (dbproto/-p-prop db :multi pid)
+                        (throw (ex-info (str "Unable to execute cas on a multi-cardinality predicate: " p-name)
+                                        {:status 400
+                                         :error  :db/validation-error})))
+
+          query'      {:select "?current-val"
```

Comment:
do we need to add on the `'` to this binding here? i didn't see an original `query` binding that this modifies.

Reply:
Fixed in update